src/resolution_functions/instrument.py

```diff
@@ -618,52 +645,125 @@ def get_model_signature(self, model_name: Optional[str] = None) -> Signature:
 
     @property
     def available_models(self) -> list[str]:
+        """
+        A list of :term:`models<model>` available for this :term:`version` of this
+        :term:`instrument`.
+
+        Only includes the recommended version of each model - does not list all versions of all
+        models.
+
+        Returns
+        -------
+        available_models
+            A list of available models.
+        """
+        return [name for name, value in self._models.items() if isinstance(value, str)]
+
+    @property
+    def all_available_models(self) -> list[str]:
         """
         A list of all :term:`models<model>` available for this :term:`version` of this
         :term:`instrument`.
 
+        Includes both all the versions of all models (see `Instrument.available_unique_models` and
+        the recommended versions (see `Instrument.available_models`).
+
         Returns
         -------
         available_models
             A list of all available models.
         """
-        return list(self._models.keys())
+        return list(self._models)
 
     @property
-    def available_models_and_configurations(self) -> dict[str, list[str]]:
+    def available_unique_models(self) -> list[str]:
         """
-        A dictionary mapping each available :term:`model` to the user
-        :term:`configurations<configuration>` available for that :term:`model`.
+        A list of all unique :term:`models<model>` available for this :term:`version` of this
+        :term:`instrument`.
 
-        All :term:`models<model>` available for this :term:`version` of this :term:`instrument`,
-        and all their :term:`configurations<configuration>` are listed.
+        Only includes the versioned models, i.e. lists all versions of all models, but does not list
+        the recommended models.
 
         Returns
         -------
-        models_and_configurations
-            All models and all their configurations.
+        available_models
+            A list of all available unique models.
         """
-        return {model_name: list(model['configurations'].keys())
-                for model_name, model in self._models.items()}
+        return [name for name, value in self._models.items() if not isinstance(value, str)]
```

Comment:
This and the function above appear identical.

Reply:
Do you mean in the code or documentation?

The intention is to provide all possibilities for listing the available models:

- `available_unique_models` (this one) for listing the actual (versioned) models - the ones with data
- `available_models` for listing the aliased models (without versions) - the ones we recommend
- `all_available_models` for listing both of the above together

Reply:
Seems fine to me (in the same order):

- ` [name for name, value in self._models.items() if not isinstance(value, str)]`
- `[name for name, value in self._models.items() if isinstance(value, str)]`
- `list(self._models)`

gives output:

- `['PyChop_fit_v1', 'PyChop_fit_v2']`
- `['PyChop_fit']`
- `['PyChop_fit', 'PyChop_fit_v1', 'PyChop_fit_v2']`

as intended. Maybe it's the difference in `not isinstance` vs `isinstance`?

Reply:
Ah! I missed the `not`, yes.